### PR TITLE
Fix incorrect handling of RPC

### PR DESCRIPTION
**Pull request by budhadityabanerjee**
Fixes issue #205 . A bad indented "else" was causing an issue with RPC's that did not have the "message-id" attribute. This fix removes that code and reworks the logic of handling the attributes.

### CODE REVIEW

File: ncclient/operations/rpc.py [hunk 1]
```diff
@@ -208,23 +208,26 @@ def callback(self, root, raw):
...
-                        # if no error delivering, can del the reference to the RPC
-                        del self._id2rpc[id]
-                        break
-        else:
-            raise OperationError("Could not find 'message-id' attribute in <rpc-reply>")
+        for key in attrs:  # in the <rpc-reply> attributes
+            message_id = attrs.get("message-id")
+            if message_id is None:
+                # required attribute so raise OperationError
+                raise OperationError("Could not find 'message-id' attribute in <rpc-reply>")
```
Comment (glennmatthews) [inline ncclient/operations/rpc.py]: This check should be done before entering the `for` loop, I think? Doesn't make much sense to repeat it for every key.

Comment (glennmatthews) [reply ncclient/operations/rpc.py]: For that matter, I'm not sure why there's a for loop here at all - it looks like message-id is the only key we care about?

Comment (glennmatthews) [reply ncclient/operations/rpc.py]: Sorry - also would `if "message-id" not in attrs` be a simpler way to perform this check?

Comment (budhadityabanerjee) [reply ncclient/operations/rpc.py]: Incorporated your comments to move the check out of the for loop and do it in a simpler way. Waiting for Einar's thoughts on removing the for loop altogether.

**Comment by einarnn** [comment]
Yes, the for loop should be removed entirely. It is only there to find the message-id, so it can be replaced.

Cheers,

Einar

**Comment by budhadityabanerjee** [comment]
Removed for loop as discussed.

**Comment by budhadityabanerjee** [comment]
Have also fixed #207 with latest commit

**Pull request merged**